streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBSegmentedBytesStore.java

```diff
@@ -67,6 +68,58 @@
     }
 
     @Override
+    public KeyValueIterator<Bytes, byte[]> all() {
+        final List<Segment> searchSpace = keySchema.allSegments(segments);
+        
+        final Segment minSegment = keySchema.minSegment(segments);
+        final Segment maxSegment = keySchema.maxSegment(segments);
+        final Bytes minKey = minSegment.first().key;
```

Comment:
Why do we need to find the min and max? We can just iterate over all items in the db, i.e., we don't need to filter.